dotCMS/src/main/java/com/dotmarketing/factories/WebAssetFactory.java

```diff
@@ -901,21 +861,37 @@ public static boolean isAbstractAsset(WebAsset asset) {
 		return false;
 	}
 
+	public static void changeAssetMenuOrder(Folder folder, int newValue, User user) throws ActionException, DotDataException {
+
+		// Checking permissions
+		if (!permissionAPI.doesUserHavePermission(folder, PERMISSION_WRITE, user))
```

Comment:
use curly brackets

Reply:
and add a log

Reply:
you can reduce this to
folder.setShowOnMenu(!(newValue == -1))

Reply:
Done